### PR TITLE
Openthread apply simplified config

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Some config options are no longer needed so I removed them to make it simpler

### CODE REVIEW

File: doc/nrf/ug_thread_configuring.rst [hunk 1]
```diff
@@ -23,40 +23,11 @@ Mandatory configuration
...
-
-* :option:`CONFIG_NET_IPV6`
-* :option:`CONFIG_NET_CONFIG_NEED_IPV6`
-
-Additionally, since Thread by default registers a considerable amount of IP addresses, the default IPv6 address count values must be increased.
-Set the following options to the provided values:
-
-* :option:`CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT` to ``6``
-* :option:`CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT` to ``8``
+  * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread will only work in NCP architecture
```
Comment (rlubos) [inline doc/nrf/ug_thread_configuring.rst:30]: Dot missing at the end of the sentence.

File: doc/nrf/ug_thread_configuring.rst [hunk 2]
```diff
@@ -23,40 +23,11 @@ Mandatory configuration
...
-
-* :option:`CONFIG_NET_IPV6`
-* :option:`CONFIG_NET_CONFIG_NEED_IPV6`
-
-Additionally, since Thread by default registers a considerable amount of IP addresses, the default IPv6 address count values must be increased.
-Set the following options to the provided values:
-
-* :option:`CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT` to ``6``
-* :option:`CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT` to ``8``
+  * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread will only work in NCP architecture.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:30]:
```suggestion
  * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread is meant to work only in the NCP architecture.
```

File: doc/nrf/ug_thread_configuring.rst [hunk 3]
```diff
@@ -88,17 +59,13 @@ It involves two devices: a Commissioner that is already in the Thread network an
 
 Configuring this process is optional, because the Thread :ref:`openthread_samples` in |NCS| use hardcoded network information.
 
-If you want to manually enable the Thread network Commissioner role on a device, set the following Kconfig options to the provided values:
+If you want to manually enable the Thread network Commissioner role on a device, set the following Kconfig option to the provided values:
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:62]:
```suggestion
If you want to manually enable the Thread network Commissioner role on a device, set the following Kconfig option to the provided value:
```

File: doc/nrf/ug_thread_configuring.rst [hunk 4]
```diff
@@ -88,17 +59,13 @@ It involves two devices: a Commissioner that is already in the Thread network an
...
 
 * :option:`CONFIG_OPENTHREAD_COMMISSIONER` to ``y``.
-* :option:`CONFIG_MBEDTLS_HEAP_SIZE` to ``10240``.
 
-To enable the Thread network Joiner role on a device, set the following Kconfig options to the provided values:
+To enable the Thread network Joiner role on a device, set the following Kconfig option to the provided values:
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:66]:
```suggestion
To enable the Thread network Joiner role on a device, set the following Kconfig option to the provided value:
```

File: doc/nrf/ug_thread_configuring.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_thread_configuring.rst:30]: Also, `NCP architecture` can link to the NCP section from https://github.com/nrfconnect/sdk-nrf/pull/2605

**Pull request merged**